vsphere/tests/test_vsphere.py

```diff
@@ -273,3 +248,35 @@ def test_service_check_ok(aggregator, instance):
                 status=VSphereCheck.OK,
                 tags=SERVICE_CHECK_TAGS
             )
+
+
+def test__instance_key(vsphere, instance):
+    assert vsphere._instance_key(instance) == "vsphere_mock"
+    del instance['name']
+    with pytest.raises(BadConfigError):
+        vsphere._instance_key(instance)
+
+
+def test__should_cache(instance):
+    now = time.time()
+    # do not use fixtures for the check instance, some params are set at
+    # __init__ time and we need to instantiate the check multiple times
+    check = VSphereCheck('vsphere', {}, {}, [instance])
+    i_key = check._instance_key(instance)
+
+    # first run should always cache
+    assert check._should_cache(instance, MORLIST) is True
+    assert check._should_cache(instance, METRICS_METADATA) is True
+
+    # simulate previous runs, set the last execution time in the past
+    init_config = {
+        'refresh_morlist_interval': REFRESH_MORLIST_INTERVAL + 1,
+        'refresh_metrics_metadata_interval': REFRESH_METRICS_METADATA_INTERVAL + 1,
```

Comment:
What's the reason behind the `+ 1` here ?

Reply:
I wanted to make sure the setting is propagated from init_config and we're not using the default, going to make this a little bit more explicit